src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/InputRightCurlyWithoutShouldStartLine.java

```diff
@@ -0,0 +1,63 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks;
+
+public class InputRightCurlyWithoutShouldStartLine{
+
+    private int a;
+    private static int b;
+    {  a = 2; }  //violation if option=ALONE
+    static { b = 3; }  //violation if option=ALONE
+
+    void method1() {
+        Thread t = new Thread() {@Override public void run() {
+            int a; int b;} //violation
+        };
+    }
+
+    void method2(java.util.HashSet<String> set) {
+        java.util.Map<String, String> map1 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");}  //violation
+        };
+
+        java.util.Map<String, String> map2 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");
+        }}; //violation
+    }
+
+    void method3() {
+        method2(new java.util.HashSet<String>() {{
+            add("XZ13s");
+            add("AB21/X");
+            add("YYLEX");
+            add("AR5E");
+        }});  //violation
+    }
+
+    int method4(int a) {
+        if (a>2) a*=10; return ++a; }   //violation
+
+    void method5(int a) {
+        while (a > 5) { a--; }  //violation if option=ALONE
+
+        if (a > 4) { a++; }  //violation if option=ALONE
+
+        do {a--;} while (a > 3);  //violation if option=ALONE
+
+        for (int i = 1; i < 10; i++) { byte b = 10; }  //violation if option=ALONE
+
+        if (a < 2) { --a; } else if (a > 3) { a++; } //2 violations if option=ALONE
+
+        java.util.List<String> list = new java.util.ArrayList<>();
+        list.stream()
+                //violation for both option=ALONE and option=ALONE_OR_SINGLELINE
+                .filter(e -> {return !e.isEmpty() && !"null".equals(e);} )
```

Comment:
Again I used a violation here since lambda wasn't the only thing that was on the line and this to my mind was an not correct for `ALONE_OR_SINGLELINE`. I feel that no more option should be added, logic can be updated if no violation for `ALONE_OR_SINGLELINE` is desired.